src/ImageSharp.Drawing/Processors/DrawPathProcessor.cs

```diff
@@ -119,19 +101,17 @@ protected override void OnApply(ImageBase<TColor> source, Rectangle sourceRectan
                 minY,
                 maxY,
                 this.ParallelOptions,
-                y =>
+                (int y) =>
                 {
                     int offsetY = y - polyStartY;
-                    var currentPoint = default(Vector2);
+
                     for (int x = minX; x < maxX; x++)
                     {
+                        // TODO add find intersections code to skip and scan large regions of this.
                         int offsetX = x - startX;
-                        currentPoint.X = offsetX;
-                        currentPoint.Y = offsetY;
+                        var info = this.Path.GetPointInfo(offsetX, offsetY);
```

Comment:
yeah the shapes library supplies the real world intersections logic the abstract classes `Region` & `Drawable`(was Path) `ImageShape.Drawing` are there as proxies to allow `SixLabors.Shapes`